monkey/monkey_island/cc/ui/src/components/report-components/security/issues/TunnelIssue.js

```diff
@@ -18,17 +18,27 @@ export function getAllTunnels(agents, machines) {
     if (!islandIPs.includes(agent.cc_server.ip)) {
       let agentMachine = getMachineByAgent(agent, machines);
       if (agentMachine !== null) {
-        let agentMachineHostname = getMachineHostname(agentMachine)
+        let agentMachineInfo = {
+          'id': agentMachine.id,
+          'ip': getMachineIPs(agentMachine)[0],
+          'hostname': getMachineHostname(agentMachine)
+        };
+
+        let agentTunnelMachineInfo = {
+          'id': -1,
+          'ip': agent.cc_server.ip,
+          'hostname': agent.cc_server.ip
+        };
 
-        let agentTunnelMachineHostname = agent.cc_server.ip;
         let agentTunnelMachine = getMachineFromIP(agent.cc_server.ip, machines)
         if (agentTunnelMachine !== null) {
-          agentTunnelMachineHostname = getMachineHostname(agentTunnelMachine);
+          agentTunnelMachineInfo.id = agentTunnelMachine.id;
+          agentTunnelMachineInfo.hostname = getMachineHostname(agentTunnelMachine);
         }
 
         tunnels.push({
-          'agent_machine': agentMachineHostname,
-          'agent_tunnel': agentTunnelMachineHostname
+          'agent_machine': agentMachineInfo,
+          'agent_tunnel': agentTunnelMachineInfo
```

Comment:
I'd use `tunnel_machine` instead, here and in the variables. The word `agent` adds no context in this case. `agent_machine` makes sense because it's the MACHINE where AGENT ran.